src/libsystemd-network/dhcp-static-lease.c

```diff
@@ -0,0 +1,141 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "dhcp-internal.h"
+#include "dhcp-server-internal.h"
+#include "memory-util.h"
+#include "strv.h"
+#include "utf8.h"
+
+#define ADDR_NULL 0
+
+static sd_dhcp_static_lease *current_static_lease = NULL;
```

Comment:
no, this is not acceptable. No static state please. See https://systemd.io/CODING_STYLE the part about code organization